src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandler.java

```diff
@@ -148,4 +150,19 @@ public Mono<RequestAuthorizationResponseDto> handle(TransactionRequestAuthorizat
                                 .doOnError(BadRequestException.class, error -> log.error(error.getMessage()))
                 );
     }
+
+    /**
+     * Convenient function that maps authorizationUrl to payment gateway
+     */
+    private String authorizationUrlToGw(String authorizationUrl) {
+        if (authorizationUrl.contains("xpay")) {
```

Comment:
This approach based on the redirection URL is not too strong and adds an implementation coupling with the redirection url configuration and this code (that will have to be updated in case of modification PGS side). 
I would rather extends the `Mono<Tuple2<String, String>> gatewayAttempts` to a `Mono<Tuple3<String, String,String>> gatewayAttempts` and put in the third tuple parameter the constant value for XPAY/VPOS/POSTEPAY based on the constructed mono

Reply:
`Mono<Tuple3<String, String,String>>` or `Mono<Tuple3<String, String,PaymentGateway>>` in case of adding enumeration for all handled payment gateways🧐